cpp/merkletree/merkle_tree.h

```diff
@@ -209,4 +209,25 @@ class MerkleTree : public cert_trans::MerkleTreeInterface {
   size_t level_count_;
 };
 
+// Mutable Merkle Tree, supports updating nodes and truncating the tree.
+class MutableMerkleTree : public MerkleTree {
+ public:
+  // The constructor takes a pointer to some concrete hash function
+  // instantiation of the SerialHasher abstract class.
+  explicit MutableMerkleTree(std::unique_ptr<SerialHasher> hasher);
+  virtual ~MutableMerkleTree();
+
+  // Update |leaf|th leaf in the tree. Indexing starts from 1.
+  // Returns false if |leaf| is out of bounds.
+  bool UpdateLeaf(size_t leaf, const std::string& data);
```

Comment:
Could we leave that one out? It's really easy to do yourself, doesn't bring much to the class (I know the parent has an equivalent, I wish to delete it too!).